tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp

```diff
@@ -0,0 +1,2789 @@
+    {
+      dumper->InsertWhitespace();
+      dumper->InsertTypeName(m_anonymousNamedStructName, m_navigationId);
+    }
+  }
+  if (dumpOptions.NeedsTrailingSemi)
+  {
+    dumper->InsertPunctuation(';');
+  }
+  if (dumpOptions.NeedsTrailingNewline)
+  {
+    dumper->Newline();
+  }
+}
+
+void AstEnum::DumpNode(AstDumper* dumper, DumpNodeOptions dumpOptions)
+{
+  if (!m_namespace.empty())
+  {
+    if (dumpOptions.NeedsNamespaceAdjustment)
+    {
+      dumper->SetNamespace(m_namespace);
+    }
+  }
+  if (dumpOptions.NeedsLeftAlign)
+  {
+    dumper->LeftAlign();
+  }
+  dumper->InsertKeyword("enum");
+  dumper->InsertWhitespace();
+  if (m_isScoped)
+  {
+    if (m_isScopedWithClass)
+    {
+      dumper->InsertKeyword("class");
+    }
+    else
+    {
+      dumper->InsertKeyword("struct");
+    }
+  }
+  dumper->InsertWhitespace();
+  dumper->InsertTypeName(m_name, m_navigationId);
+  if (m_isFixed)
+  {
+    dumper->InsertWhitespace();
+    dumper->InsertPunctuation(':');
+    dumper->InsertWhitespace();
+    dumper->InsertMemberName(m_underlyingType);
+  }
+  dumper->Newline();
+  dumper->LeftAlign();
+  dumper->InsertPunctuation('{');
+  dumper->AdjustIndent(2);
+  dumper->Newline();
+  bool firstEnumerator = true;
+  for (auto const& enumerator : m_enumerators)
+  {
+    if (!firstEnumerator)
+    {
+      dumper->InsertPunctuation(',');
+      dumper->Newline();
+    }
+    firstEnumerator = false;
+    enumerator->DumpNode(dumper, dumpOptions);
+  }
+  dumper->Newline();
+  dumper->AdjustIndent(-2);
+  dumper->LeftAlign();
+  dumper->InsertPunctuation('}');
+  if (dumpOptions.NeedsTrailingSemi)
+  {
+    dumper->InsertPunctuation(';');
+  }
+  if (dumpOptions.NeedsTrailingNewline)
+  {
+    dumper->Newline();
+  }
+}
+
+void AstField::DumpNode(AstDumper* dumper, DumpNodeOptions dumpOptions)
+{
+  if (dumpOptions.NeedsLeftAlign)
+  {
+    dumper->LeftAlign();
+  }
+  m_fieldType.Dump(dumper, dumpOptions);
+  dumper->InsertWhitespace();
+  dumper->InsertMemberName(m_name);
+  // if (m_initializer)
+  //{
+  //   DumpNodeOptions innerOptions{dumpOptions};
+  //   if (m_classInitializerStyle == ICIS_CopyInit)
+  //   {
+  //     dumper->InsertWhitespace();
+  //     dumper->InsertPunctuation('=');
+  //     dumper->InsertWhitespace();
+  //   }
+  //   else if (m_classInitializerStyle == ICIS_ListInit)
+  //   {
+  //     innerOptions.DumpListInitializer = true;
+  //   }
+  //   m_initializer->Dump(dumper, innerOptions);
+  // }
+  if (dumpOptions.NeedsTrailingSemi)
+  {
+    dumper->InsertPunctuation(';');
+  }
+  if (dumpOptions.NeedsTrailingNewline)
+  {
+    dumper->Newline();
+  }
+}
+
+void AstType::Dump(AstDumper* dumper, DumpNodeOptions dumpOptions) const
+{
+  //  if (m_isConstQualified)
+  //  {
+  //    dumper->InsertKeyword("const");
+  //    dumper->InsertWhitespace();
+  //  }
+  //  if (m_underlyingType)
+  //  {
+  //    m_underlyingType->Dump(dumper);
+  //  }
+  //  else
```

Comment:
```

Reply:
One of my major shortcuts was punting on type formatting. I need to revisit this because it results in some incorrect (and excessively verbose) outputs in the ApiView. This code was from before I punted on AstType.